python/cudf/cudf/core/frame.py

```diff
@@ -573,6 +582,18 @@ def equals(self, other, **kwargs):
         else:
             return self._index.equals(other._index)
 
+    def _explode(self, explode_column_num: int, index: Optional[cudf.Index]):
```

Comment:
Instead of explicitly passing the index here, what if we passed the `ignore_index` option down into this function? Then we can check `if self.index is not None` in case we're exploding an Index / MultiIndex object for example.